crates/secret-service-proto/src/v1/wire.rs

```diff
@@ -25,34 +25,26 @@ use super::{
 #[allow(clippy::large_enum_variant)]
 pub enum ServerMessage {
     /// The message the client sent was invalid.
-    InvalidClientMessage,
+    InvalidClientMessage(String),
 
     /// The server experienced an unexpected internal error while handling the
     /// request.
     ///
     /// Check the server logs for debugging details.
     OpaqueServerError,
 
-    /// Response for [`WalletSigner::sign`](super::traits::WalletSigner::sign).
-    GeneralWalletSign {
-        /// Schnorr signature for a certain message.
-        sig: [u8; 64],
-    },
-
-    /// Response for [`WalletSigner::pubkey`](super::traits::WalletSigner::pubkey).
-    GeneralWalletPubkey {
-        /// Serialized Schnorr [`XOnlyPublicKey`](bitcoin::XOnlyPublicKey) for operator signatures.
-        pubkey: [u8; 32],
-    },
+    /// An explicit signal from the the server that the client should immediately retry the request
+    TryAgain,
 
-    /// Response for [`WalletSigner::sign`](super::traits::WalletSigner::sign).
-    StakechainWalletSign {
+    /// Response for [`SchnorrSigner::sign`](super::traits::SchnorrSigner::sign) and
+    /// [`SchnorrSigner::sign_no_tweak`](super::traits::SchnorrSigner::sign_no_tweak)
+    WalletSignerSign {
         /// Schnorr signature for a certain message.
         sig: [u8; 64],
```

Comment:
Replace the 64 for https://docs.rs/secp256k1/0.29.1/secp256k1/constants/constant.SCHNORR_SIGNATURE_SIZE.html